highfidelity/hifi
# Case 19409, Case 19468: Fading fixes and shader cleanup

https://highfidelity.manuscript.com/f/cases/19409/Fading-avatars-shadows-don-t-fade
https://highfidelity.manuscript.com/f/cases/19468/Avatars-are-never-marked-as-not-fading

Test plan:
- Be in a zone with shadows.  Have another avatar enter the domain you are in so that they appear in front of you.  Both they and their shadow should fade in (in master, the shadow would just appear without fading).  Have them leave and they and their shadow should both fade out.  Have them come back and both should fade in again.
- Run the debugHighlight.js script.  Turn on your bubble.  Hover over the other avatar.  Walk in to them so they disappear and then back up so they reappear, but keep your mouse pointed at where they should be.  As they fade in, the highlight should only cover the fading parts.
- Run the material tests: https://github.com/highfidelity/hifi_tests/tree/master/tests/engine/render/material
